examples/single-page-app/envoy.yml

```diff
@@ -0,0 +1,307 @@
+                pass_through_matcher:
+                  name: ":path"
+                  string_match:
+                    safe_regex:
+                      regex: >-
+                        ^\/(authorize.*|hub.*|login|logout)$
+                  invert_match: true
+                redirect_path_matcher:
+                  path:
+                    prefix: /authorize
+                signout_path:
+                  path:
+                    exact: /logout
+                credentials:
+                  client_id: "0123456789"
+                  token_secret:
+                    name: token
+                    sds_config:
+                      resource_api_version: V3
+                      path_config_source:
+                        path: /etc/envoy/secrets/myhub-token-secret.yml
+                  hmac_secret:
+                    name: hmac
+                    sds_config:
+                      resource_api_version: V3
+                      path_config_source:
+                        path: /etc/envoy/secrets/hmac-secret.yml
+                auth_scopes:
+                - user:email
+          - name: envoy.filters.http.compressor
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.compressor.v3.Compressor
+              response_direction_config:
+                common_config:
+                  min_content_length: 100
+                disable_on_etag_header: true
+              compressor_library:
+                name: text_optimized
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip
+                  memory_level: 3
+                  window_bits: 10
+          - name: envoy.router
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.router.v3.Router
+          codec_type: "AUTO"
+          stat_prefix: ingress_http
+          route_config:
+            virtual_hosts:
+            - name: web
+              domains: ["*"]
+              routes:
+              - match:
+                  prefix: "/hub/"
+                route:
+                  host_rewrite_literal: api.myhub
+                  regex_rewrite:
+                    pattern:
+                      regex: '^/hub/(.*)'
+                    substitution: '/\1'
+                  cluster: hub-api
+              - match:
+                  prefix: "/"
+                route:
+                  cluster: loopback
+      transport_socket:
+        name: envoy.transport_sockets.tls
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext
+          common_tls_context:
+            tls_certificates:
+            # The following self-signed certificate pair is generated using:
+            # $ openssl req -x509 -newkey rsa:2048 -keyout a/front-proxy-key.pem -out  a/front-proxy-crt.pem -days 3650 -nodes -subj '/CN=front-envoy'
```

Comment:
Strongly dislike inlining certs in configuration files, even if examples. I would prefer we move these to distinct files. If we have to regenerate at any point it will be easier, it makes the config easier to read and also avoids an anti-pattern of putting private keys in configuration, which we don't want to use as an exemplar.

Reply:
the problem is that without doing this it requires generating the keys and then adding related docs

that just isnt the focus of the sandbox - its more here as guidance for next steps

in this case it would really break the flow of the docs - the cert would need to be generated at the beginning before the composition could come up

how about we add a comment explaining this is an anti-pattern for docs only - and then do the same for all the other instances of it ?

Reply:
for ref

```console
$ git grep "The following self"
examples/brotli/envoy.yaml:            # The following self-signed certificate pair is generated using:
examples/brotli/envoy.yaml:            # The following self-signed certificate pair is generated using:
examples/front-proxy/envoy.yaml:            # The following self-signed certificate pair is generated using:
examples/tls/envoy-https-http.yaml:            # The following self-signed certificate pair is generated using:
examples/tls/envoy-https-https.yaml:            # The following self-signed certificate pair is generated using:
examples/websocket/envoy-wss.yaml:            # The following self-signed certificate pair is generated using:
examples/win32-front-proxy/front-envoy.yaml:            # The following self-signed certificate pair is generated using:
examples/zstd/envoy.yaml:            # The following self-signed certificate pair is generated using:
examples/zstd/envoy.yaml:            # The following self-signed certificate pair is generated using:
```

Reply:
Why can't these certs be checked in like our other example certs, but referenced on the file system? The end user doesn't need to regenerate them themselves.

That allows easier substitution with prod examples and regeneration if we want to change for whatever reason, expiry, project name change, whatever.

It can be a separate GitHub issue, but this strikes me as an anti-pattern for users.

Reply:
lets deal with this separately

Reply:
OK, please file an issue for this one, I think this one is worth tracking.

Reply:
added here https://github.com/envoyproxy/envoy/issues/31664

Reply:
Thanks! Will resolve.